src/rsdp.rs

```diff
@@ -29,11 +32,16 @@ impl RsdpV1Tag {
         str::from_utf8(&self.signature).ok()
     }
 
-    /// The value to add to all the other bytes (of the Version 1.0 table) to calculate the Checksum of the table.
-    ///
-    /// If this value added to all the others and casted to byte isn't equal to 0, the table must be ignored. 
-    pub fn checksum(&self) -> u8 {
-        self.checksum
+    /// Validation of the RSDPv1 checksum
+    pub fn checksum_is_valid(&self) -> bool {
+        let result = unsafe {
+            let bytes = slice::from_raw_parts(
+                self as *const _ as *const u8,
+                RSDPV1_LENGTH + 8
+            );
+            bytes[8..].iter().fold(0usize, |acc, val| acc.wrapping_add(*val as usize))
+        };
+        (result & 0xff) == 0
```

Comment:
Do you have a reference for why you're summing into a `usize` and then masking for the lowest byte later? My interpretation of the ACPI spec is that you should sum within a single byte:
```
This is the checksum of the fields defined in the ACPI 1.0 specification. This includes only the first 20 bytes of this table, bytes 0 to 19, including the checksum field. These bytes must sum to zero.
```

This would apply to both `checksum_is_valid` methods:
```suggestion
            bytes[8..].iter().fold(0u8, |acc, val| acc.wrapping_add(*val))
        };
        result == 0
```

Reply:
yeah all we need is a u8 and that would remove the need for the mask later. I think in the original version i didn't use wrapping add, so the use of usize was to avoid an overflow.

Reply:
Updated